web3/_utils/caching/caching_utils.py

```diff
@@ -142,16 +173,60 @@ def is_cacheable_request(
 }
 
 INTERNAL_VALIDATION_MAP: Dict[
-    "RPCEndpoint", Callable[[SYNC_PROVIDER_TYPE, Sequence[Any], Dict[str, Any]], bool]
+    "RPCEndpoint",
+    Callable[
+        [SYNC_PROVIDER_TYPE, Sequence[Any], Dict[str, Any]],
+        bool,
+    ],
 ] = {
     **{endpoint: always_cache_request for endpoint in ALWAYS_CACHE},
-    **{endpoint: validate_blocknum_in_params for endpoint in BLOCKNUM_IN_PARAMS},
-    **{endpoint: validate_blocknum_in_result for endpoint in BLOCKNUM_IN_RESULT},
-    **{endpoint: validate_blockhash_in_params for endpoint in BLOCKHASH_IN_PARAMS},
+    **{endpoint: validate_from_block_id_in_params for endpoint in BLOCKNUM_IN_PARAMS},
+    **{endpoint: validate_from_blocknum_in_result for endpoint in BLOCK_IN_RESULT},
+    **{endpoint: validate_from_blockhash_in_params for endpoint in BLOCKHASH_IN_PARAMS},
 }
 CACHEABLE_REQUESTS = tuple(INTERNAL_VALIDATION_MAP.keys())
 
 
+def set_threshold_if_empty(provider: SYNC_PROVIDER_TYPE) -> None:
+    current_threshold = provider.request_cache_validation_threshold
+
+    if current_threshold is empty or isinstance(
+        current_threshold, RequestCacheValidationThreshold
+    ):
+        try:
+            # turn off momentarily to avoid recursion
+            provider.cache_allowed_requests = False
+            chain_id_result = provider.make_request("eth_chainId", [])["result"]
+            chain_id = int(chain_id_result, 16)
+
+            if (
+                isinstance(
+                    current_threshold,
+                    RequestCacheValidationThreshold,
+                )
+                and chain_id != 1
+            ):
+                provider.logger.debug(
+                    "Request cache validation threshold is set to "
+                    f"{current_threshold.value} "
+                    f"for chain with chain_id `{chain_id}` but this value only works "
+                    "on chain_id `1`. Setting to default value for chain_id "
```

Comment:
Good catch. It should work with others as well. I think this messaging should be improved. Will fix 👍🏼

Reply:
I think we can just remove that validation entirely. I was trying to issue a warning related to that value not being available, for friendliness, but that responsibility may have to be on the user since we can't realistically keep track of all chains that have it available.